File: SEARCHING/MEDIUM/Single Element in a Sorted Array/Code2.py
# PROBLEM LINK:- https://leetcode.com/problems/single-element-in-a-sorted-array/


class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        low = 0
        high = len(nums) - 2
        while low <= high:
            mid = low + (high - low)//2
            
            if nums[mid] == nums[mid^1]:
                low = mid + 1
            else:
                high = mid - 1
                
        return nums[low]

File: SEARCHING/MEDIUM/Single Element in a Sorted Array/Code2.cpp
/*
PROBLEM LINK:- https://leetcode.com/problems/single-element-in-a-sorted-array/
*/

class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        if(nums.size() == 0)
            return nums[0];
        
        int low = 0, high = nums.size() - 1, mid, x;
        
        while(low < high)
        {
            mid = low + (high - low)/2;
            x = (mid % 2 == 0) ? mid+1 : mid-1;
            
            if(nums[mid] == nums[x])
                low = mid + 1;
            else
                high = mid;
        }
        
        return nums[low];
    }
};
